salt/modules/useradd.py

```diff
@@ -187,6 +192,8 @@ def add(name,
             except OSError:
                 # /etc/usermgmt.conf not present: defaults will be used
                 pass
+    if usergroup == False:
```

Comment:
the default for 'usergroup' is None. Should you compare this to None too? I'm thinking to change this to: if not usergroup:

Reply:
There are three possibilities - we explicitly set `usergroup` to `False` to trigger `-N` to require no user group be created, we set it to `True` to trigger `-U` to require a user group to be created, or don't touch it at all which will go with the OS defaults - which differs depending on the distribution or user preferences. 

If you instead go with `if not usergroup`, you'll be force overriding the OS defaults (or overriding whatever the user configured the OS to do), so I imagine most people won't be too happy about that.

Reply:
Ah, okay. So it's intended to use the OS defaults unless specified.

Reply:
Shouldn't we then document this so that a user knows when to set this to `False` and what behavior that will produce?

Reply:
@rares-pop Correct. I don't see any reason to change things the user didn't request. @cachedout Yep, fair call. Will do.